langchain/document_loaders/web_base.py

```diff
@@ -6,19 +6,41 @@
 from langchain.docstore.document import Document
 from langchain.document_loaders.base import BaseLoader
 
+header_template = {
+    "User-Agent": "",
+    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,*/*"
+    ";q=0.8",
+    "Accept-Language": "en-US,en;q=0.5",
+    "Referer": "https://www.google.com/",
+    "DNT": "1",
+    "Connection": "keep-alive",
+    "Upgrade-Insecure-Requests": "1",
+}
+
 
 class WebBaseLoader(BaseLoader):
     """Loader that uses urllib and beautiful soup to load webpages."""
 
     def __init__(self, web_path: str):
         """Initialize with webpage path."""
         self.web_path = web_path
+        self.session = requests.Session()
+
+        try:
+            from fake_useragent import UserAgent
+
+            header_template["User-Agent"] = UserAgent().random
+            self.session.headers = dict(header_template)
```

Comment:
can we add an option to pass this during __init__? just to make it easy for users to customize

Reply:
Good call.  will push soon